orangecontrib/network/widgets/OWNxFromDistances.py

```diff
@@ -248,11 +248,13 @@ def generateGraph(self, N_changed=False):
             # exclude unconnected
             if self.node_selection != NodeSelection.ALL_NODES:
                 n_components, components = csgraph.connected_components(edges)
-                counts = np.bincount(components, minlength=n_components + 1)
+                counts = np.bincount(components)
                 if self.node_selection == NodeSelection.COMPONENTS:
-                    mask = counts >= self.excludeLimit
+                    ind = [i for i, c in enumerate(counts) if
+                           c >= self.excludeLimit]
```

Comment:
This should be faster - for large sparsely connected graphs:

```
ind = np.nonzero(counts >= self.excludeLimit)[0]
```

Reply:
Or `ind = np.flatnonzero(counts >= self.excludeLimit)`. (I keep forgetting there also exists `flatnonzero` :(.)

Reply:
Fixed.